webapp/app/routes.py

```diff
@@ -317,6 +316,18 @@ def error_500(error):
             'An uncaught error occurred', exc_info=error.original_exception)
         return render_template('error/500.html', header_title=_('500.header-title'), js_needed=False), 500
 
+    @app.errorhandler(EricaRequestTimeoutError)
+    def error_500(error):
+        current_app.logger.error(
+            'An Erica Request Timeout error occurred', exc_info=error.original_exception)
+        return render_template('error/500.html', header_title=_('500.header-title'), js_needed=False), 500
```

Comment:
It's okay for me to return the 500.html template. But we should maybe use 504 as HTTP code instead of 500. 

https://airbrake.io/blog/http-errors/504-gateway-timeout-error

We can then also catch it better in the client. What do you think?

Reply:
Yes, this was my idea too.  We can then also use custom message in template if usefull.